File: src/token.zig
pub const TokenType = enum {
    H1,
    H2,
    H3,
    H4,
    H5,
    H6,
    P,
    Code,
    Link,
    Quote,
    Image,
    NewLine,
    Illegal,
    Ol,
    Ul,

    pub fn header_level(level: usize) TokenType {
        return switch (level) {
            1 => .H1,
            2 => .H2,
            3 => .H3,
            4 => .H4,
            5 => .H5,
            else => .H6,
        };
    }
};

pub const Token = struct {
    const Self = @This();

    token_type: TokenType,
    content: []const u8,

    pub fn new(tok_type: TokenType, content: []const u8) Token {
        return Self{ .token_type = tok_type, .content = content };
    }
};

File: src/token.rs
#[derive(Debug, PartialEq, Eq, Copy, Clone)]
pub enum TokenType {
    H1,
    H2,
    H3,
    H4,
    H5,
    H6,
    P,
    Code,
    Link,
    Quote,
    Image,
    Illegal,
    Ol,
    Ul,
    Eof,
}

#[derive(Debug, PartialEq, Eq)]
pub struct Token {
    ttype: TokenType,
    content: String,
}

impl Token {
    pub fn new(ttype: TokenType, content: String) -> Self {
        Self { ttype, content }
    }

    pub fn ttype(&self) -> TokenType {
        self.ttype
    }

    pub fn content(&self) -> String {
        self.content.clone()
    }
}

impl TokenType {
    pub fn header_level(level: &str) -> TokenType {
        match level {
            "#" => TokenType::H1,
            "##" => TokenType::H2,
            "###" => TokenType::H3,
            "####" => TokenType::H4,
            "#####" => TokenType::H5,
            _ => TokenType::H6,
        }
    }
}
